nomad/core_sched_test.go

```diff
@@ -439,26 +441,68 @@ func TestCoreScheduler_EvalGC_Batch_OldVersion(t *testing.T) {
 	alloc2.DesiredStatus = structs.AllocDesiredStatusRun
 	alloc2.ClientStatus = structs.AllocClientStatusLost
 
-	// Insert alloc with older job modifyindex
-	alloc3 := mock.Alloc()
-	job2 := job.Copy()
+	err = store.UpsertAllocs(structs.MsgTypeTestSetup, jobModifyIdx+2, []*structs.Allocation{alloc, alloc2})
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
 
-	alloc3.Job = job2
-	alloc3.JobID = job2.ID
+	// Insert allocs with indexes older than job.ModifyIndex. Two cases:
```

Comment:
For clarity, because of what we'll actually check:

```suggestion
	// Insert allocs with indexes older than job.ModifyIndex and job.JobModifyIndex. Two cases:
```